mcs/class/System/System.IO.Compression/DeflateStream.cs

```diff
@@ -104,8 +104,13 @@ internal DeflateStream (Stream stream, CompressionLevel compressionLevel, bool l
 		{
 		}		
 
+		~DeflateStream () => Dispose (false);
+
 		protected override void Dispose (bool disposing)
 		{
+			if (disposing)
+				GC.SuppressFinalize (this);
```

Comment:
Wouldn't this go in the next conditional? The logic around the disposing is hard for me to follow, so maybe I'm misunderstanding.

Reply:
@CoffeeFlux it's called "dispose pattern", see https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose#implementing-the-dispose-pattern-for-a-base-class